src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -89,25 +89,58 @@ def deployment_validate_table_format(result):
     return result
 
 
+def _populate_alternate_kwargs(kwargs):
+    """ Translates the parsed arguments into a format used by generic ARM commands
+    such as the resource and lock commands. """
+    parent = ''
+    has_child = all(kwargs[x] is not None for x in ['child_name', 'child_type'])
+    has_grandchild = all(kwargs[x] is not None for x in ['grandchild_name', 'grandchild_type'])
+    resource_namespace = kwargs['namespace']
+    resource_type = kwargs['grandchild_type'] or kwargs['child_type'] or kwargs['type']
+    resource_name = kwargs['grandchild_name'] or kwargs['child_name'] or kwargs['name']
+    if has_grandchild:
+        parent = '{type}/{name}/providers/{child_namespace}/{child_type}/{child_name}/providers/{grandchild_namespace}'.format(**kwargs)  # pylint: disable=line-too-long
+    elif has_child:
+        parent = '{type}/{name}/providers/{child_namespace}'.format(**kwargs)
+    parent = parent.replace('providers/None', '')
```

Comment:
It isn't a valid resource provider, which is why remove it. It is VERY possible. For example:

`providers/Microsoft.Network/virutalNetworks/vnet1/subnets/subnet1`

Would resolve to a child_namespace of None, which is where you get "providers/None". It's an artifact of the regex and the fact that child resources can belong to the same provider as the parent or a different one.